src/main/java/org/jabref/gui/actions/ExportLinkedFilesService.java

```diff
@@ -0,0 +1,125 @@
+package org.jabref.gui.actions;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+
+import org.jabref.Globals;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.util.OptionalUtil;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class ExportLinkedFilesService extends Service<List<CopyFilesResultViewModel>> {
```

Comment:
I'm not sure but wasn't a service intended to be reused while a task is for single work items? I think `Task` would fit better here.

Reply:
Yeah, I did not really know the difference